### DISCUSSION TITLE
Stashing a value to use across multiple tests is not returning a value

### DISCUSSION THREAD

**Discussion by MeganOzanne** [pactumjs/pactum#264] [General]
Hi, I'm trying to build out some tests using Pactum and my final hurdle is passing my auth token across multiple files so that I don't need to call my auth endpoint for every test. Please could someone assist me to see what I'm doing wrong?

My setup is using:
Pactum
Cucmber
Typescript
Chai

My request to get my auth token is this:
` export async function getAccessToken() {
    
        const username = user
        const password = p
  
       await spec()
        .post('auth.url')
          .withBody({
              grant_type: Auth0.GrantType,
              realm: 'All-Passwordless-Users',
              username,
              password,
              audience: Auth0.Audience,
              scope: auth0Scope,
              client_id: auth0ClientId,
          })
         .expectStatus(200)
         .stores('accessToken', this.response.body.access_token)
  //this.response = await s.toss()
 //return this.response.body.access_token
 //.returns((ctx) => { return ctx.res.body.access_token });
     
};`

I then attempt to use ` .withHeaders('Authorization', '$S{GetAuthToken}')` within my further requests, within different files but it outputs $S{GetAuthToken} rather then the required value.

I have also tried using:
`const getToken = getAccessToken().then(
  function(accessToken) {return accessToken;}
);

const token = await getToken`

But this returns `TypeError: getAccessToken is not a function`

Given Pactum provides a way to store and retrieve data I thought this task would be fairly simple. TIA

**Comment by leelaprasadv** [comment]
Hi @MeganOzanne,

Since you are using CucumberJS, you can add the code to retrieve access token in a `BeforeAll` / `Before` hook and use PactumJS `setDefaultHeaders()` once in the same hook to set the Authorization header as default for all subsequent requests. 

You can still leverage `stores` and reference the store variable to set the headers.

```js
request.setDefaultHeaders('Authorization', '$S{accessToken}');
```

Reference: 
https://pactumjs.github.io/api/settings/setDefaultHeaders.html#setdefaultheaders

Here is an example cucmberJS implementation with PactumJS (doesn't have auth setup included and not in typescript though)
 https://github.com/pactumjs/pactum-cucumber-boilerplate

**Reply by MeganOzanne** [reply]
Thanks Leela, the trouble I am now hitting is when I attempt to log out the token with `console.log('using:', '$S{accessToken}')` it returns: using: $S{accessToken} rather then the value.

How do I log out what is stored so I can see if this is the cause of the error I am now facing?

**Reply by leelaprasadv** [reply]
You can use `parse` to retrieve the data. 

https://pactumjs.github.io/api/utils/parse.html#parse

**Reply by MeganOzanne** [reply]
Thank you
